mm2src/mm2_main/src/lp_healthcheck.rs

```diff
@@ -453,30 +456,48 @@ mod tests {
     cross_test!(test_valid_message, {
         let ctx = ctx();
         let message = HealthcheckMessage::generate_message(&ctx, false).unwrap();
-        assert!(message.is_received_message_valid());
+        message.is_received_message_valid().unwrap();
     });
 
     cross_test!(test_corrupted_messages, {
         let ctx = ctx();
 
         let mut message = HealthcheckMessage::generate_message(&ctx, false).unwrap();
-        message.data.expires_at += 1;
-        assert!(!message.is_received_message_valid());
+        message.data.expires_at_secs += healthcheck_message_exp_secs() * 3;
+        assert_eq!(
+            discriminant(&message.is_received_message_valid().err().unwrap()),
+            discriminant(&SignValidationError::LifetimeOverflow {
+                max_allowed_expiration_secs: 0,
+                remaining_expiration_secs: 0
+            })
+        );
```

Comment:
nit: maybe `assert!(matches!(something, SignValidationError::LifetimeOverflow { .. }))`

Reply:
I prefer to keep the explicit use of `discriminants` because there is no way to break it and it's literally made for this purpose. The behavior of `matches` can vary when structs used with fields instead of `{ .. }`.